app/src/main/java/com/example/topmovies/viewmodel/MovieViewModel.kt

```diff
@@ -64,30 +63,25 @@ class MovieViewModel constructor(
             EnumScreen.FAVORITE ->
                 removeMovieFromFavorites(id)
         }
-        favoriteMovies.update()
     }
 
     private fun addMovieToFavorites(id: String) {
-        val mutableMovies = movies.value!!.toMutableList()
-        val movie = movies.value!!.find { movie -> movie.id.contains(id) }
-        val copy = movie?.copy(isFavorite = true)
-        if (copy != null) {
-            mutableMovies[movies.value!!.indexOf(movie)] = copy
-            movies.postValue(mutableMovies)
-            _favoriteMovies.add(copy)
-            favoritePref.edit().putString(copy.id, "").apply()
+        repository.updateMovie(id, true)
+        repository.getMovies {
+            handler.post { movies.value = it }
+        }
+        repository.getFavoriteMovie {
+            handler.post { favoriteMovies.value = it }
         }
     }
 
     private fun removeMovieFromFavorites(id: String) {
-        val mutableMovies = movies.value!!.toMutableList()
-        val movie = movies.value!!.find { movie -> movie.id.contains(id) }
-        val copy = movie?.copy(isFavorite = false)
-        if (copy != null) {
-            mutableMovies[movies.value!!.indexOf(movie)] = copy
-            movies.postValue(mutableMovies)
-            _favoriteMovies.remove(movie)
-            favoritePref.edit().remove(copy.id).apply()
+        repository.updateMovie(id, false)
+        repository.getMovies {
+            handler.post { movies.value = it }
+        }
+        repository.getFavoriteMovie {
+            handler.post { favoriteMovies.value = it }
```

Comment:
Same here.
From the logical end, why would you need to get everything from the database?
Why not update the same movie? Why should we re-fetch everything?